app/storage/datastore.py

```diff
@@ -0,0 +1,83 @@
+from flask import current_app
+from google.cloud import datastore
+from structlog import get_logger
+
+from app.data_model import app_models
+from app.storage.errors import ItemAlreadyExistsError
+
+logger = get_logger()
+
+TABLE_CONFIG = {
+    app_models.SubmittedResponse: {
+        'key_field': 'tx_id',
+        'table_name_key': 'EQ_SUBMITTED_RESPONSES_TABLE_NAME',
+        'schema': app_models.SubmittedResponseSchema,
+    },
+    app_models.QuestionnaireState: {
+        'key_field': 'user_id',
+        'table_name_key': 'EQ_QUESTIONNAIRE_STATE_TABLE_NAME',
+        'schema': app_models.QuestionnaireStateSchema,
+    },
+    app_models.EQSession: {
+        'key_field': 'eq_session_id',
+        'table_name_key': 'EQ_SESSION_TABLE_NAME',
+        'schema': app_models.EQSessionSchema,
+    },
+    app_models.UsedJtiClaim: {
+        'key_field': 'jti_claim',
+        'table_name_key': 'EQ_USED_JTI_CLAIM_TABLE_NAME',
+        'schema': app_models.UsedJtiClaimSchema,
+    },
+}
```

Comment:
I'm not sure what the solution is, but it feels wrong to have the properties specific to each model - `key_field` and `schema` in the storage class and not in the app_models.

Reply:
It's not ideal.  I think much of the schema stuff is fairly redundant where the backends essentially take JSON as input already.  Potentially some of it could be taken out entirely, but didn't want to make too many changes in one go.

Reply:
Yeah, I'm happy for a separate card to be created to review the use of the schemas.